### ISSUE TITLE
Enter Wifi Credentials Page is not opening

### ISSUE THREAD

**Issue by rubenfeurer** [rubenfeurer/internetRadio#58]
When the Pi is in AP mode, after I connect to it via Wifi, the «Wifi Settings» is not opening up. Therefore I'm not able to add the Wifi Credentials.

See Video: https://share.icloud.com/photos/092HPQU_kWttIfVjBRX7UWslA

**Comment by AmnaNaeem91** [comment]
I will check why it is not opening.

Scenario is you are connected to hotspot and it doesn't allow you to open
this page for settings the WiFI credentials?

**Comment by rubenfeurer** [comment]
> Scenario is you are connected to hotspot and it doesn't allow you to open

Correct. It should automatically open the Wifi Setting page when connected to wifi.

**Comment by AmnaNaeem91** [comment]
This page is opening, issue is of IP. the IP visible in terminal is
hardcoded in print statement, I guess it is not the actual IP address

**Comment by AmnaNaeem91** [comment]
I have updated code to display actual IP address, please check them

**Comment by rubenfeurer** [comment]
Now when in AP mode and connecting to it via Wifi I'm able to open http://raspberrypi.local:8080/ to change the Wifi credentials. However, still http://raspberrypi.local:8080/wifi-setup is not opening automatically after connecting to Pi via Wifi in AP mode. Any ideas how to fix? I think it was working before no?

**Comment by AmnaNaeem91** [comment]
I guess it was not working, because I don't know how that is done, I told
you that after version 1, because that on receiving side browser which I
can't modify

**Comment by AmnaNaeem91** [comment]
What I did was if in AP mode, even if you open home page it takes you wifi
settings page. That I did in code from pi side

**Comment by rubenfeurer** [comment]
Do you think the following code could work?
https://www.perplexity.ai/search/is-it-possible-to-automaticall-zDALlg85QfqjolRWVSMUiw

**Comment by AmnaNaeem91** [comment]
I couldn't find any code just the perplexity empty page with search bar

**Comment by rubenfeurer** [comment]
To automatically open a webpage when connecting to WiFi using Python, you can implement a solution that involves creating a **captive portal**. This is commonly used in public WiFi networks to redirect users to a specific webpage upon connection. Below are the key steps and components involved in achieving this:

## Requirements

1. **Access Point Support**: Your router must support captive portals. This is essential for redirecting users automatically when they connect to your WiFi network[1].
2. **Python Environment**: You will need Python installed on your device, along with libraries such as `Flask` for creating a web server and `scapy` or similar for network management.

## Steps to Create a Captive Portal

### 1. Set Up the Access Point

You need to configure your router or device to act as an access point. This can often be done through the router's settings or by using software that allows for custom firmware (like OpenWRT).

### 2. Create a Simple Web Server

Using Python, you can set up a simple web server that serves your desired HTML page. Here’s a basic example using Flask:

```python
from flask import Flask, redirect

app = Flask(__name__)

@app.route('/')
def home():
    return redirect("http://yourdesiredwebpage.com", code=302)

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=80)
```

### 3. Handle Network Requests

You can use libraries like `scapy` to monitor connections and detect when a new device connects to your WiFi network. When a device connects, you can trigger the redirection to your web server.

### 4. Configure DNS and DHCP

To ensure that all requests are redirected to your web server, configure the DNS settings of your access point to point to the local IP address of your server. This often involves setting up DHCP options on your router.

### 5. Test Your Setup

Once everything is configured, connect a device to your WiFi network and check if it redirects automatically to the specified webpage.

## Additional Considerations

- **Security**: Ensure that your captive portal is secure and does not expose sensitive information.
- **User Experience**: Design the landing page to provide clear instructions or information for users.
- **Compliance**: Be aware of legal implications regarding user data collection and privacy when implementing captive portals.

By following these steps, you can successfully create an environment where connecting to your WiFi network automatically opens a specified webpage using Python[1][2][3].

Citations:
[1] https://stackoverflow.com/questions/66058349/how-to-create-wifi-splash-page-using-python
[2] https://github.com/OpenAgricultureFoundation/python-wifi-connect
[3] https://www.pythoncentral.io/how-to-connect-wi-fi-using-python/
[4] https://forums.raspberrypi.com/viewtopic.php?t=173157
[5] https://www.youtube.com/watch?v=SzYKzAHsdMg
[6] https://dev.to/fernand0/please-allow-me-to-control-my-router-from-a-python-script-1ik6
[7] https://gist.github.com/a740122/f37c990284e67dbd138b079447d5d748
